apps/backend/src/app/modules/maps/maps.service.ts

```diff
@@ -1891,13 +1907,13 @@ export class MapsService {
       throw new BadRequestException('BSP is not compressed');
   }
 
-  async getBspFromTemp(userID: number): Promise<File> {
+  async getBspFromTemp(userID: number): Promise<File | Error> {
     const userIDObjects = await this.fileStoreService.listFileKeys(
       `upload_tmp/${userID}`
     );
 
     if (userIDObjects.length === 0) {
-      throw new BadRequestException('Missing BSP file');
+      return new Error('Missing BSP file');
```

Comment:
As above, let's return `null` here, not an error (which are typically reserved for `try/catch`)